src/SFA.DAS.LearnerDataMismatches.IntegrationTests/TestConfigurationExtensions.cs

```diff
@@ -20,5 +20,27 @@ public static void AddPages(this ServiceCollection services)
             services.RemoveAll(typeof(T));
             services.AddScoped(service);
         }
+
+        public static void ConfigureMockServices<T1, T2>(
```

Comment:
I would write only one method that can be fluently appended multiple times for as many services are required. 
```
public static ServiceCollection ConfigureMockService<T>(
            this ServiceCollection services,
            Func<IServiceProvider, T> service)
            where T : class
        {
            services.RemoveAll(typeof(T));
            services.AddScoped(service);
            return services;
        }
```
I think if you use this, you will not need these overrides. You can use this for multiple services as such 
```
private static ServiceCollection CreateServices()
    {
        var services = new ServiceCollection();
        new Startup(configuration).ConfigureServices(services);
        services.AddLogging();
        services.AddPages();
        services
         .ConfigureMockServices(_ => CommitmentsApi)
         .ConfigureMockServices(_ => ProviderApi)
         .ConfigureMockServices(_ => AccountsApi);
        return services;
    }
```

Reply:
Looking at it now, I prefer this way of doing it.  Originally I wanted to write

`services.ConfigureMockServices(CommitmentsApi, ProviderApi, AccountsApi)`

However they need to be provided as functors, which immediately pushes them onto many lines.  And in that case you are right, the fluent style is cleaner.